### DISCUSSION TITLE
How to install u-boot script

### DISCUSSION THREAD

**Discussion by ghuband** [rauc/meta-rauc#238] [Q&A]
I have a system that uses u-boot with partitions for boot and rootfs:
bootA  rootfsA  bootB rootfsB
bootA and bootB contain the device tree file (oftree), the kernel file (zImage), and a u-boot script file (boot.scr).
I see that bundle.bbclass can install the kernel file (RAUC_SLOT_kernel) and the device tree file (RAUC_SLOT_dtb), but how do I install the u-boot script file?

**Comment by ejoerns** [comment]
RAUC installs images or tar archives, this means it can install a kernel into a dedicated kernel partition, a devicetree into a devicetree partition, etc.
If you have a shared partition for multiple artifacts, just put them into the same image. The meta-rauc convenience for kernel and device tree type would not help here.

As an alternative I would suggest to compile boot.scr into the u-boot and load kernel, dtb and rootfs together from the same partition. Then you could also use what U-boot refers to as "distro boot": https://u-boot.readthedocs.io/en/latest/develop/bootstd.html

**Reply by mdsketch** [reply]
Any chance you have more guidance on how to do this? I believe this is the behaviour I want, where I can update kernel, dtb and rootfs all in 1 partition for simplicity. I currently have dtb and kernel  on root fs, but I can't figure out how to update uboot

**Reply by anthonio9** [reply]
Same issue here, would really appreciate more cues on this.

**Reply by ejoerns** [reply]
Which information do you miss exactly? Information on how to configure U-Boot?

**Reply by anthonio9** [reply]
Yes, exactly that. I'm not that familiar with U-boot configuration, I'm starting to read about it now.
